app/ante/utils_test.go

```diff
@@ -1,104 +1,109 @@
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 
 	"github.com/cosmos/ethermint/app"
 	ante "github.com/cosmos/ethermint/app/ante"
 	"github.com/cosmos/ethermint/crypto/ethsecp256k1"
 	ethermint "github.com/cosmos/ethermint/types"
 	evmtypes "github.com/cosmos/ethermint/x/evm/types"
 
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+
 	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 
-	abci "github.com/tendermint/tendermint/abci/types"
-	tmcrypto "github.com/tendermint/tendermint/crypto"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 )
 
 type AnteTestSuite struct {
 	suite.Suite
 
-	ctx         sdk.Context
-	app         *app.EthermintApp
-	anteHandler sdk.AnteHandler
+	ctx            sdk.Context
+	app            *app.EthermintApp
+	encodingConfig params.EncodingConfig
+	anteHandler    sdk.AnteHandler
 }
 
 func (suite *AnteTestSuite) SetupTest() {
 	checkTx := false
 
 	suite.app = app.Setup(checkTx)
-	suite.app.Codec().RegisterConcrete(&sdk.TestMsg{}, "test/TestMsg", nil)
 
-	suite.ctx = suite.app.BaseApp.NewContext(checkTx, abci.Header{Height: 1, ChainID: "ethermint-3", Time: time.Now().UTC()})
+	suite.ctx = suite.app.BaseApp.NewContext(checkTx, tmproto.Header{Height: 2, ChainID: "ethermint-3", Time: time.Now().UTC()})
+	suite.app.AccountKeeper.SetParams(suite.ctx, authtypes.DefaultParams())
 	suite.app.EvmKeeper.SetParams(suite.ctx, evmtypes.DefaultParams())
 
-	suite.anteHandler = ante.NewAnteHandler(suite.app.AccountKeeper, suite.app.EvmKeeper, suite.app.SupplyKeeper)
+	suite.encodingConfig = app.MakeEncodingConfig()
+	// We're using TestMsg amino encoding in some tests, so register it here.
+	suite.encodingConfig.Amino.RegisterConcrete(&testdata.TestMsg{}, "testdata.TestMsg", nil)
+
+	suite.anteHandler = ante.NewAnteHandler(suite.app.AccountKeeper, suite.app.BankKeeper, suite.app.EvmKeeper, suite.encodingConfig.TxConfig.SignModeHandler())
 }
 
 func TestAnteTestSuite(t *testing.T) {
 	suite.Run(t, new(AnteTestSuite))
 }
 
-func newTestMsg(addrs ...sdk.AccAddress) *sdk.TestMsg {
-	return sdk.NewTestMsg(addrs...)
+func newTestMsg(addrs ...sdk.AccAddress) *testdata.TestMsg {
+	return testdata.NewTestMsg(addrs...)
 }
 
 func newTestCoins() sdk.Coins {
 	return sdk.NewCoins(ethermint.NewPhotonCoinInt64(500000000))
 }
 
-func newTestStdFee() auth.StdFee {
-	return auth.NewStdFee(220000, sdk.NewCoins(ethermint.NewPhotonCoinInt64(150)))
+func newTestStdFee() legacytx.StdFee {
+	return legacytx.NewStdFee(220000, sdk.NewCoins(ethermint.NewPhotonCoinInt64(150)))
 }
 
 // GenerateAddress generates an Ethereum address.
-func newTestAddrKey() (sdk.AccAddress, tmcrypto.PrivKey) {
+func newTestAddrKey() (sdk.AccAddress, cryptotypes.PrivKey) {
 	privkey, _ := ethsecp256k1.GenerateKey()
 	addr := ethcrypto.PubkeyToAddress(privkey.ToECDSA().PublicKey)
 
 	return sdk.AccAddress(addr.Bytes()), privkey
 }
 
 func newTestSDKTx(
```

Comment:
is there a reason why we use the legacy types for this but we are using `sdk.Tx` as the new type (looks to be replacing auth.StdTx which is now a legacy type)

Reply:
this may affect the results of the test later on if we use legacy signatures and such to test the new type (?)